tests/NuGet.IndexingTests/CamelCaseFilterTests.cs

```diff
@@ -0,0 +1,165 @@
+                        new TokenAttributes("AaBb", 0, 4, 0),
+                        new TokenAttributes("Bb", 2, 4, 1),
+                        new TokenAttributes("BbCc", 2, 6, 0),
+                        new TokenAttributes("Cc", 4, 6, 1),
+                        new TokenAttributes("CcDd", 4, 8, 0),
+                        new TokenAttributes("Dd", 6, 8, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "AaBbCc",
+                    new[]
+                    {
+                        new TokenAttributes("AaBbCc", 0, 6, 1),
+                        new TokenAttributes("Aa", 0, 2, 0),
+                        new TokenAttributes("AaBb", 0, 4, 0),
+                        new TokenAttributes("Bb", 2, 4, 1),
+                        new TokenAttributes("BbCc", 2, 6, 0),
+                        new TokenAttributes("Cc", 4, 6, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "AaBb",
+                    new[]
+                    {
+                        new TokenAttributes("AaBb", 0, 4, 1),
+                        new TokenAttributes("Aa", 0, 2, 0),
+                        new TokenAttributes("Bb", 2, 4, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "Aa",
+                    new[]
+                    {
+                        new TokenAttributes("Aa", 0, 2, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    string.Empty,
+                    new object[0]
+                };
+
+                yield return new object[]
+                {
+                    "ABCD",
+                    new[]
+                    {
+                        new TokenAttributes("ABCD", 0, 4, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "ABCde",
+                    new[]
+                    {
+                        new TokenAttributes("ABCde", 0, 5, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "AaB",
+                    new[]
+                    {
+                        new TokenAttributes("AaB", 0, 3, 1),
+                        new TokenAttributes("Aa", 0, 2, 0),
+                        new TokenAttributes("B", 2, 3, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "Abcd",
+                    new[]
+                    {
+                        new TokenAttributes("Abcd", 0, 4, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "A1a2B3b",
```

Comment:
Add comments to describe that this is the current behavior, but something that we might want to tweak in the future. So this doesn't get interpreted as spec